core/Pos/Core/Fee.hs

```diff
@@ -0,0 +1,90 @@
+module Pos.Core.Fee
+       ( Coeff(..)
+       , TxSizeLinear(..)
+       , calculateTxSizeLinear
+       , TxFeePolicy(..)
+       ) where
+
+import           Universum
+
+import qualified Data.Aeson                 as JSON
+import qualified Data.Aeson.Types           as JSON
+import           Data.Fixed                 (Fixed (..), Nano, resolution, showFixed)
+import qualified Data.Text.Buildable        as Buildable
+import           Formatting                 (bprint, build, shown, (%))
+import           Serokell.Data.Memory.Units (Byte, toBytes)
+import qualified Data.HashMap.Strict        as HM.S
+
+-- | A fractional coefficient of fixed precision.
+newtype Coeff = Coeff Nano
+    deriving (Eq, Show, Generic, NFData)
+
+instance Buildable Coeff where
+    build (Coeff x) = fromString (showFixed True x)
+
+instance JSON.FromJSON Coeff where
```

Comment:
Also I just realized there **is** is a good reason why we didn't define aeson instances for core types. That's because some of them should be transferred to frontend and they may need different serialization formats. So middleware guys are kings of JSON serialization currently.